jaxley/utils/cell_utils.py

```diff
@@ -291,9 +294,11 @@ def interpolate_xyz(loc: float, coords: np.ndarray):
     Return:
         Interpolated xyz coordinate at `loc`, shape `(3,).
     """
-    return vmap(lambda x: jnp.interp(loc, jnp.linspace(0, 1, len(x)), x), in_axes=(1,))(
```

Comment:
`jnp.interp` takes `x`, `xp` and `fp`. The crux is that in the current implementation `xp=jnp.linspace(0, 1, len(x)), x)`. This would mean every `fp` i.e. the xyz coords along which we want to interpolate are equidistant (i.e. the pathlengths between them are the same), which they usually are not!

Reply:
Ohhh I see! Thanks!